WooCommerce MCP integration: v0.3.0 asset still reports 0.1.0 and adapter abilities not registered (breaks /woocommerce/mcp)

Hi team,

I’m trying to follow the WooCommerce MCP integration docs and ran into what looks like a version / packaging problem with the `mcp-adapter` plugin that prevents Woo’s MCP server from working.

## Environment

- Hosting: Pressable
- WordPress: 6.8.3
- WooCommerce: 10.3.5
- Abilities API plugin: 0.4.0 (`abilities-api` active)
- MCP Adapter: from the GitHub release tagged `v0.3.0`
- Woo MCP Beta toggle: enabled in **WooCommerce → Settings → Advanced → Features**

## What I did

1. Went to the `mcp-adapter` repo releases page and downloaded the `mcp-adapter.zip` asset from the v0.3.0 release (not the “Source code” zip).
2. Installed it via **Plugins → Add New → Upload Plugin**.
3. WordPress reported I was “updating 0.1.0 to 0.1.0”:
   - Plugin header in `mcp-adapter.php` still says:  
     `Version: 0.1.0`
   - The “Current” and “Uploaded” versions in the WP update screen both show `0.1.0`.
4. Activated the plugin and enabled MCP Beta in WooCommerce settings.
5. Installed and activated the Abilities API plugin (0.4.0).

## Expected

- MCP Adapter should register its own abilities with the Abilities API:
  - `mcp-adapter/discover-abilities`
  - `mcp-adapter/get-ability-info`
  - `mcp-adapter/execute-ability`
- WooCommerce MCP server should be available at:
  - `/wp-json/woocommerce/mcp`
  as described in the WooCommerce MCP documentation.

## Actual

1. Error log fills with messages like:

   ```text
   [ERROR] WordPress ability 'mcp-adapter/discover-abilities' does not exist. | Context: ["RegisterAbilityAsMcpTool::mcp-adapter\/discover-abilities"]
   [ERROR] WordPress ability 'mcp-adapter/get-ability-info' does not exist. | Context: ["RegisterAbilityAsMcpTool::mcp-adapter\/get-ability-info"]
   [ERROR] WordPress ability 'mcp-adapter/execute-ability' does not exist. | Context: ["RegisterAbilityAsMcpTool::mcp-adapter\/execute-ability"]


This suggests the adapter’s own abilities are not being registered with the Abilities API.

From WordPress, the REST route index (/wp-json) shows MCP routes like:

/mcp

/mcp/mcp-adapter-default-server

/wp/v2/wpmcp (from a separate MCP-for-Woo plugin I tested)

but no /woocommerce/mcp route at all.

Hitting https://example.com/wp-json/woocommerce/mcp returns:

{"code":"rest_no_route","message":"No route was found matching the URL and request method.","data":{"status":404}}


which is consistent with the route never being registered.

From CLI, the plugin header still reports 0.1.0:

grep -m1 'Version:' wp-content/plugins/mcp-adapter/mcp-adapter.php
# → Version: 0.1.0


So it looks like the v0.3.0 GitHub release asset is shipping a plugin that still identifies itself as 0.1.0 in the header, and (based on the logs) may not be registering its internal abilities correctly with abilities-api.

Impact

Because the adapter’s abilities are missing, WooCommerce’s MCP integration never reaches a “healthy” state, and the /woocommerce/mcp endpoint described in the Woo docs is not available. That means I can’t actually work with WooCommerce MCP from MCP clients (Claude + @automattic/mcp-wordpress-remote) as documented.

Request

Can you confirm whether the mcp-adapter.zip asset on the v0.3.0 release is correct?

If it is the intended build, could the plugin header version be bumped to match the tag (e.g. Version: 0.3.0) and the registration of the mcp-adapter/* abilities verified against Abilities API 0.4.0 and WooCommerce 10.3.5?

If there is a newer or different package I should be using with the Woo MCP Beta, a pointer in the README / docs would be very helpful.

Thanks for all the work on this project! I'm looking forward to working with this MCP!  I'm happy to provide more logs or CLI output if that would help reproduce the issue.

--- a/includes/Core/McpAdapter.php
+++ b/includes/Core/McpAdapter.php
@@ -24,7 +24,7 @@ use WP\MCP\Servers\DefaultServerFactory;
  */
 final class McpAdapter {
 
-	public const VERSION = '0.1.0';
+	public const VERSION = '0.3.1';
 
 	/**
 	 * Registry instance

--- a/mcp-adapter.php
+++ b/mcp-adapter.php
@@ -12,7 +12,7 @@
  * Plugin URI:        https://github.com/WordPress/mcp-adapter
  * Description:       Adapter for Abilities API, letting the abilities to be used as MCP tools, resources or prompts.
  * Requires at least: 6.8
- * Version:           0.1.0
+ * Version:           0.3.1
  * Requires PHP:      7.4
  * Author:            WordPress.org Contributors
  * Author URI:        https://github.com/WordPress/mcp-adapter/graphs/contributors
@@ -40,7 +40,7 @@ function constants(): void {
 	/**
 	 * Version of the plugin.
 	 */
-	define( 'WP_MCP_VERSION', '0.1.0' );
+	define( 'WP_MCP_VERSION', '0.3.1' );
 }
 
 constants();

